ledger/src/blockstore.rs

```diff
@@ -549,7 +549,14 @@ impl Blockstore {
 
     /// Determines if starting_slot and ending_slot are connected
     pub fn slots_connected(&self, starting_slot: Slot, ending_slot: Slot) -> bool {
-        let mut next_slots: VecDeque<_> = vec![starting_slot].into();
+        if starting_slot == ending_slot {
+            return true;
+        }
```

Comment:
One corner case I just thought of is if someone is starting from genesis (with no snapshot, starting slot will be 0 when this is called) and they're trying to create a snapshot at slot 0. In this case, we do need slot 0 to be present. Having special logic around genesis ins't great, but not the only place we do this; https://github.com/solana-labs/solana/issues/25577 discusses a way we could mitigate this for future clusters.

Reply:
Wondering if that edge-case is worth adding a special case for. The purpose of this function is to more quickly find if we're missing a slot we need so we don't waste 10-15 minutes. If we're replaying from genesis and need slot 0, though this function would pass, we're still not going to waste time because it'd fail at slot 0.

It might be more correct to only call this function iff we have a snapshot slot though

Reply:
> If we're replaying from genesis and need slot 0, though this function would pass, we're still not going to waste time because it'd fail at slot 0

Good point, if we're making a snapshot at slot `s`, the check would verify that we have `(0, s]` and then we would start with slot 0.

> It might be more correct to only call this function iff we have a snapshot slot though

Yeah, I guess it basically boils down to that the `is_full` check should be exclusive on start_bound when loading from snapshot, and inclusive when loading from genesis. But, the check does provide value in both the start-from-genesis and start-from-snapshot cases.

Given your point that we'd fail fast with slot 0 not being full despite the check saying they're connected, I think I'd be happy with covering this with comments. What do you think about elaboration with `slots_connected()` to explicitly state that `starting_slot` is excluded from `is_full()` check?

Reply:
Added some clarifying comments, lmk what you think.